src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/OOPTagHelperResolver.cs

```diff
@@ -100,33 +101,17 @@ protected virtual async Task<TagHelperResolutionResult> ResolveTagHelpersOutOfPr
             // when it's disconnected (user stops the process).
             //
             // This will change in the future to an easier to consume API but for VS RTM this is what we have.
-            try
-            {
-                var remoteClient = await RazorRemoteHostClient.CreateAsync(_workspace, CancellationToken.None);
-
-                var args = new object[]
-                {
-                    projectSnapshot,
-                    factory?.GetType().AssemblyQualifiedName,
-                };
+            var remoteClient = await RazorRemoteHostClient.CreateAsync(_workspace, CancellationToken.None);
 
-                var result = await remoteClient.TryRunRemoteAsync<TagHelperResolutionResult>(
-                    "GetTagHelpersAsync",
-                    workspaceProject.Solution,
-                    args,
-                    CancellationToken.None).ConfigureAwait(false);
+            var factoryTypeName = factory?.GetType().AssemblyQualifiedName;
 
-                return result.HasValue ? result.Value : null;
-            }
-            catch (Exception ex)
-            {
-                // We silence exceptions from the OOP host because we don't want to bring down VS for an OOP failure.
-                // We will retry all failures in process anyway, so if there's a real problem that isn't unique to OOP
-                // then it will report a crash in VS.
-                _errorReporter.ReportError(ex, projectSnapshot);
-            }
+            var result = await remoteClient.TryInvokeAsync<IRemoteLanguageService, TagHelperResolutionResult>(
+                workspaceProject.Solution,
+                (service, solutionInfo, cancellationToken) => service.GetTagHelpersAsync(solutionInfo, projectSnapshot, factoryTypeName, cancellationToken),
```

Comment:
Here we need to either make ProjectSnapshot MessagePack-serializable or convert it to ProjectSnapshotHandle.

Reply:
Ya all our serializer does today IIRC is translate a snapshot into a handle